design/one-pager-component-mutable-and-versioning.md

```diff
@@ -252,16 +264,18 @@ spec:
                 revision: frontend-c8bb659c5 
 ```
 
-Once the `frontend` component has been changed, ApplicationConfiguration will trigger
+Once the `frontend` component changed, the `ApplicationConfiguration` controller will trigger
 an upgrade for the application.
 
-1. Create a new workload with the same name of the new component revision.
-2. Check all traits' traitDefinition.
-   - Update all traits who have no revision aware to point to the new workload.
-   - If there are traits who have revision aware and specify to old revision, make sure the workload
+1. Create or update the workload to the new component revision according to revision field of binding traits.
+   - if creating new workload, workload name will align with revision name.
+   - if updating workload, workload name will align with the component name.
```

Comment:
so the workload name will change during updating?

Reply:
yes, according to trait